Formula/q/qemu.rb

```diff
@@ -103,36 +103,23 @@ def install
-    assert_match expected, shell_output("#{bin}/qemu-system-alpha --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-arm --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-cris --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-hppa --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-i386 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-m68k --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-microblaze --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-microblazeel --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-mips --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-mips64 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-mips64el --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-mipsel --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-nios2 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-or1k --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-ppc --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-ppc64 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-riscv32 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-riscv64 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-rx --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-s390x --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-sh4 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-sh4eb --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-sparc --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-sparc64 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-tricore --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-x86_64 --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-xtensa --version")
-    assert_match expected, shell_output("#{bin}/qemu-system-xtensaeb --version")
+    archs = %w[
+      aarch64 alpha arm cris hppa i386 m68k microblaze microblazeel mips
+      mips64 mips64el mipsel nios2 or1k ppc ppc64 riscv32 riscv64 rx
+      s390x sh4 sh4eb sparc sparc64 tricore x86_64 xtensa xtensaeb
+    ]
+    archs.each do |guest_arch|
+      assert_match expected, shell_output("#{bin}/qemu-system-#{guest_arch} --version")
+    end
+
     resource("homebrew-test-image").stage testpath
     assert_match "file format: raw", shell_output("#{bin}/qemu-img info FLOPPY.img")
+
+    # On macOS, verify that we haven't clobbered the signature on the qemu-system-x86_64 binary
+    if OS.mac?
+      output = shell_output("codesign --verify --verbose #{bin}/qemu-system-x86_64 2>&1")
```

Comment:
What about aarch64?

Reply:
Are you suggesting that `qemu-system-aarch64` should also be codesigned? So far this doesn't seem to be the case, at least for me on Intel:

```
$ codesign --verify --verbose $(which qemu-system-x86_64)
/usr/local/bin/qemu-system-x86_64: valid on disk
/usr/local/bin/qemu-system-x86_64: satisfies its Designated Requirement
```

```
$ codesign --verify --verbose $(which qemu-system-aarch64)
/usr/local/bin/qemu-system-aarch64: code object is not signed at all
In architecture: x86_64
```

I would expect both to be signed on Apple Silicon, since Homebrew codesigns (ad-hoc) all binaries on Apple Silicon (but I would expect tests in brew itself to catch a regression in that case, if it were widespread to affect all binaries).

Reply:
For ARM Mac, `qemu-system-aarch64` is expected to be signed (and `qemu-system-x86_64` doesn't need to be signed).

Reply:
I can adjust the test to check the binary depending on target arch, but again, on ARM Mac, Homebrew already forcibly ad-hoc codesigns everything.